### ISSUE TITLE
Broken window focus 

### ISSUE THREAD

**Issue by apodtele** [git-for-windows/git#4360]
When I launch a program, e.g. [ftgamma](https://github.com/ubawurinna/freetype-windows-binaries/blob/master/demos/win64/ftgamma.exe) which opens a window, the focus stays on the mintty window. It used to switch to the launched window as it should normally, as it does from cmd. Something broke in 2.40.0.I know that it is not git but I have no idea which component either.

**Comment by xiezhipeng-git** [comment]
install git 64 windows on windows11 .then open git bash.This will cause all programs to lose focus on windows11. You can only turn off gitbash.
filename Git-2.40.0-64-bit.exe
This is an extremely serious bug. Please repair it as soon as possible

**Comment by rimrul** [comment]
I can't reproduce this issue.

**Comment by xiezhipeng-git** [comment]
> I can't reproduce this issue.

from github release download git.then install Git-2.40.0-64-bit.exe and Git-2.39.2-64-bit.exe all defualt.then open gitbash.then move other programs   （two programs .the second and later.） all programs to lose focus on windows11

**Comment by apodtele** [comment]
@rimrul

Please try ftgamma, which is a simple program with nothing beyond CreateWindow and some basic GDI, or you can try something else. I just confirmed that it used to work and grab the keyboard focus under Windows 10. The most recent version under Windows 11 does not.

Notepad is not a good example to try. I re-filed this issue with [mintty](https://github.com/mintty/mintty/issues/1206).

**Comment by apodtele** [comment]
_pbrush.exe_ also starts in the back without focus, but mintty also loses focus while staying raised. This is Windows 11 fully updated on the brand new Dell desktop. I have a feeling that this has something to do with popup/notification features.

**Comment by mintty** [comment]
https://github.com/mintty/mintty/issues/1206#issuecomment-1481858258

**Comment by apodtele** [comment]
To narrow it down, this seems to be a problem on Windows 11 only. To reproduce, see if you can start _pbrush.exe_ (aka _mspaint.exe_) from the command line normally. For me it starts behind the mintty window with the focus lost completely for either window.

**Comment by rimrul** [comment]
`pbrush.exe` starts in front of the `Git Bash` Window for me and has the keyboard focus (pressing `Ctrl+O` opens the expected file open dialog).

**Comment by mintty** [comment]
@rimrul, is that on Windows 11?

**Comment by rimrul** [comment]
Yes.

**Comment by apodtele** [comment]
Windows `calc` is another example: `mintty` holds the input focus while `calc` blinks pink in the Taskbar.

**Comment by mintty** [comment]
Thanks for the `calc` test case which, again, gets the focus for me on Windows 10.

**Comment by apodtele** [comment]
Perhaps I will post a video. 

https://github.com/git-for-windows/git/assets/32550896/14a5c7f3-0510-430f-8637-d9578c5dc44d

**Comment by tmssngr** [comment]
I can reproduce this for `calc`, `fasmw` (flat assembler editor) or `notepad++`.

**Comment by mintty** [comment]
I had meanwhile tested this on Windows 11. Running calc from a cygwin shell in mintty transfers focus to calc for me.

**Comment by boogisha** [comment]
Just wanted to report that I have this issue now as well (upgraded to Windows 11 last night).

Doing a simple thing like `git commit` opens Notepad++ window in the background, behind git-bash (MINGW64), requiring me to switch to it manually first before typing in the commit message.

For whatever that is worth, I have the editor setup like this (working fine on Windows 10 for years):
`core.editor="C:\\Program Files (x86)\\Notepad++\\notepad++.exe" -multiInst -notabbar -nosession -noPlugin`

Using `git version 2.51.0.windows.1`

**Comment by dscho** [comment]
Does this work as expected when you run Git Bash from the Windows Terminal?

**Comment by apodtele** [comment]
There are interesting observations when wrapping the executable into a shell script, https://github.com/mintty/mintty/issues/1206#issuecomment-2910183493 and down. The issue could be with _bash_.

**Comment by boogisha** [comment]
> Does this work as expected when you run Git Bash from the Windows Terminal?

@dscho Yes, when I do `git commit --amend` from there (for example), Notepad++ comes up on top, as expected.

For the reference, this is a tutorial I found for running Git Bash from the Windows Terminal (note it says to point to `Git\bin\bash.exe` instead of `Git\git-bash.exe`, but I assume that is correct, being what I did, too).

> To run Git Bash from the Windows Terminal, you need to add it as a profile within the Windows Terminal settings.
> 1. Open Windows Terminal Settings:
> Open Windows Terminal.
> Click the down arrow next to the open tabs and select "Settings" (or press Ctrl + ,).
> 2. Add a New Profile:
> In the Settings window, navigate to the "Profiles" section.
> Click "Add new profile" or the "+" icon to create a new profile.
> Choose "New empty profile."
> 3. Configure the Git Bash Profile:
> Name: Enter a descriptive name, such as "Git Bash."
> Command line: Browse to the location of bash.exe within your Git installation directory. This is typically C:\Program Files\Git\bin\bash.exe.
> Starting directory: Optionally, set a default starting directory for Git Bash sessions.
> Icon: Browse to the git-for-windows.ico file, usually found in C:\Program Files\Git\mingw64\share\git\git-for-windows.ico, to give your Git Bash profile its distinct icon.
> 4. Save and Use:
> Click "Save" to apply the changes.
> Now, when you open a new tab in Windows Terminal, you can select "Git Bash" from the dropdown menu to launch a Git Bash session.

**Comment by boogisha** [comment]
> (note it says to point to `Git\bin\bash.exe` instead of `Git\git-bash.exe`, but I assume that is correct, being what I did, too)

Indeed, just tried changing to `Git\git-bash.exe`, and that just opens git-bash (MINGW64) as a separate window/app (not inside Windows Terminal).

**Comment by dscho** [comment]
You could always use this option when installing Git for Windows:

<img width="571" height="477" alt="Image" src="https://github.com/user-attachments/assets/35238727-b693-4d94-b89d-a80ff3de93cc" />

**Comment by boogisha** [comment]
Is there any (more or less significant) difference between using Git-Bash and Git Bash from Windows Terminal? Any drawback using the latter...?

I see colors are not exactly the same, but they're pretty much similar, so that should be fine. I also highly prefer font (and size) inside Git-Bash, but I guess it's a matter of configuring Windows Terminal to match (or getting used to it).

**Comment by dscho** [comment]
> Is there any (more or less significant) difference between using Git-Bash and Git Bash from Windows Terminal? Any drawback using the latter...?

There are no significant differences between them. The only reason why Git Bash in MinTTY is the default in Git for Windows is that Windows Terminal did not exist back in 2007 when the Git for Windows project was started, and neither in 2015 when it migrated from MSys to MSYS2 (and migrated Git Bash from the ConHost window to MinTTY).

Windows Terminal is not installed by default on Windows 8.1 or Windows 10 (which are both [still supported by Git for Windows](https://gitforwindows.org/requirements)), therefore it would not be possible to change Git for Windows' default everywhere, but https://github.com/git-for-windows/git/issues/5807 does suggest to change the default where possible.
